Makefile

```
@@ -1,15 +1,18 @@
-TARGETS=$(patsubst dates/%.yaml, dates/%/ALL, $(wildcard dates/*.yaml))
-ALL: ${TARGETS}
+GENERATOR='./generate.rb'
+
+MAKE_TARGETS=$(patsubst dates/%.yaml, dates/%/ALL, $(wildcard dates/*.yaml))
+CLEAN_TARGETS=$(patsubst dates/%.yaml, dates/%/*, $(wildcard dates/*.yaml))
+ALL: ${MAKE_TARGETS}
 
 clean:
-	rm -rf ${TARGETS}
+	rm -rf ${CLEAN_TARGETS}
 
 # Keep Makefiles
-.SECONDARY: $(patsubst dates/%/ALL, dates/%/Makefile, $(TARGETS))
+.SECONDARY: $(patsubst dates/%/ALL, dates/%/Makefile, $(MAKE_TARGETS))
 
 dates/%/Makefile: dates/%.yaml $(wildcard decks/*.yaml)
 	mkdir -p $(shell dirname $@)
-	./generate.rb $< $(shell dirname $@)
+	$(GENERATOR) $< $(shell dirname $@) || printf "\033[1;31minterpreter or script missing\033[0m\n"
```

Is it intended to ignore the error here?

---

it was itended to make the user aware of the reason as it might be used of people not familiar with Make, LaTeX nor Ruby or alike.

---

```patch
+   RUBY := ruby

    dates/%/Makefile: dates/%.yaml $(wildcard decks/*.yaml)
         mkdir -p $(shell dirname $@)
-       ./generate.rb $< $(shell dirname $@)
+       $(RUBY) ./generate.rb $< $(shell dirname $@) \
+       || (printf "\033[1;31Please install Ruby!\033[0m\n"; exit 1)
```

---

when does it fail while ruby is installed? (I do not know)

---

Let's say the ruby script exit due an Exception or whatever reason

```
target:
    ruby -e "raise 'Oh no!'" || printf "\033[1;31minterpreter or script missing\033[0m\n"
```

``` sh
make: Entering directory '/etc/nixos/nixpkgs'
ruby -e "raise 'Oh no!'" || printf "\033[1;31minterpreter or script missing\033[0m\n"
-e:1:in `<main>': Oh no! (RuntimeError)
interpreter or script missing
make: Leaving directory '/etc/nixos/nixpkgs'
```

---

@vv01f Do you agree that this is a good change and can you update your pull request?

---

@farao ack, will update

---

updated and dropped coloing as it didn't work any more on my machine and also ate first char of the string